src/logger/src/lib.rs

```diff
@@ -32,3 +32,17 @@ fn extract_guard<G>(lock_result: LockResult<G>) -> G {
         Err(poisoned) => poisoned.into_inner(),
     }
 }
+
+pub fn update_metric_with_elapsed_time(
+    metric: &SharedMetric,
+    start_time_us: u64,
+    action_to_log: &str,
+    thread_name: &str,
```

Comment:
Instead of passing in thread name as `&str` just use [ThreadId](https://doc.rust-lang.org/std/thread/struct.ThreadId.html) in the debug message.